sdk/communication/AzureCommunicationCommon/Tests/ObjC/ObjCAutoRefreshProactiveTokenCredentialTests.m

```diff
@@ -0,0 +1,296 @@
+                                                 NSError * _Nullable error) {
+            XCTAssertNotNil(accessToken);
+            XCTAssertNil(error);
+            XCTAssertEqual(accessToken.token, tokenValidFor15Mins);
+            XCTAssertEqual(weakSelf.fetchTokenCallCount, 0);
+            [expectation fulfill];
+        }];
+    });
+    [self waitForExpectationsWithTimeout:self.timeout handler:^(NSError * _Nullable error) {
+        [self failForTimeout:error testName:expectation.expectationDescription];
+    }];
+}
+
+-(CommunicationTokenCredential *)createTokenRefreshOptions:(__weak ObjCAutoRefreshProactiveTokenCredentialTests *)weakSelf initialToken:(NSString *)initialToken refreshedToken:(NSString *) refreshedToken {
+    return (CommunicationTokenCredential *) [[CommunicationTokenRefreshOptions alloc]
+            initWithInitialToken:initialToken
+            refreshProactively:YES
+            tokenRefresher:^(void (^ block)
+                             (NSString * _Nullable accessToken,
+                              NSError * _Nullable error)) {
+                weakSelf.fetchTokenCallCount += 1;
+                block(refreshedToken, nil);
+            }];
+}
+
+- (void)test_ShouldBeCalledImmediatelyWithExpiredToken {
+    XCTestExpectation *expectation = [self expectationWithDescription:@"test_ShouldBeCalledImmediatelyWithExpiredToken"];
+    
+    __weak ObjCAutoRefreshProactiveTokenCredentialTests *weakSelf = self;
+    CommunicationTokenRefreshOptions *tokenRefreshOptions = [self createTokenRefreshOptions:weakSelf initialToken:self.sampleExpiredToken refreshedToken:weakSelf.sampleToken];
+    
+    CommunicationTokenCredential *credential = [[CommunicationTokenCredential alloc]
+                                                initWithOptions: tokenRefreshOptions
+                                                error:nil];
+    dispatch_after(dispatch_time(DISPATCH_TIME_NOW, (int64_t)(1 * NSEC_PER_SEC)), dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_HIGH, 0), ^{
+        [credential tokenWithCompletionHandler:^(CommunicationAccessToken * _Nullable accessToken,
+                                                 NSError * _Nullable error) {
+            XCTAssertNotNil(accessToken);
+            XCTAssertNil(error);
+            XCTAssertEqual(accessToken.token, weakSelf.sampleToken);
+            XCTAssertEqual(weakSelf.fetchTokenCallCount, 1);
+            [expectation fulfill];
+        }];
+    });
+    [self waitForExpectationsWithTimeout:self.timeout handler:^(NSError * _Nullable error) {
+        [self failForTimeout:error testName:expectation.expectationDescription];
+    }];
+}
+
+- (void)test_ShouldBeCalledIfTokenExpiringSoon {
+    XCTestExpectation *expectation = [self expectationWithDescription:@"test_ShouldBeCalledIfTokenExpiringSoon"];
+    
+    dispatch_semaphore_t semaphore = dispatch_semaphore_create(0);
+    int minutes[] = {1, 9};
```

Comment:
I suppose anything above 9 is covered by `test_ShouldNotBeCalledBeforeExpiringTime`.

Reply:
yes